### PR TITLE
Convert markdown content in LinkedIn

### PR DESCRIPTION
**Pull request by arash77**
Same as #58

### CODE REVIEW

File: lib/plugins/linkedin.py [hunk 1]
```diff
@@ -47,16 +49,24 @@ def wrap_text_with_index(self, content):
...
-            [f"#{v}" for v in hashtags]
-        )  # the mentions are not linked to anyone!
+        hashtags = " ".join([f"#{v}" for v in hashtags])
         if len(images) > 20:
             warnings = f"A maximum of 20 images, not {len(images)}, can be included in a single linkedin post."
             images = images[:20]
         else:
             warnings = ""
 
+        # convert markdown formatting because Mastodon doesn't support it
```
Comment (bgruening) [inline lib/plugins/linkedin.py:61]: should this be a shared function in both classes?

Comment (arash77) [reply lib/plugins/linkedin.py]: Should I create a function in `galaxy_social.py` to be used by plugins?

**Comment by wm75** [comment]
I guess it's time for a client base class that ships as a default plugin and defines the method?

**Comment by wm75** [comment]
That class could also raise NotImplemented for format_content and create_post, i.e. enforce that derived client classes define these methods.

**Comment by arash77** [comment]
So should I just revert the change to have a simple code for the conversion of LinkedIn content and work on the client base class later?

**Comment by wm75** [comment]
Sounds good from my side, yes.

**Comment by arash77** [comment]
Ready for review @wm75

**Comment by wm75** [comment]
Had an idea for a simpler implementation, but you've worked out everything that's required @arash77 !

**Pull request closed without merging**